### PR TITLE
 [Az.RecoveryServices.Backup] fixed sql CRR and security issue with SQL restore

### PR DESCRIPTION
**Pull request by hiaga**
## Description

fixed security issue with SQL restore
fixed Get AAD Properties for SQL CRR

## Checklist

- [x] I have read the [_Submitting Changes_](../blob/master/CONTRIBUTING.md#submitting-changes) section of [`CONTRIBUTING.md`](../blob/master/CONTRIBUTING.md)
- [x] The title of the PR is clear and informative
- [x] The appropriate `ChangeLog.md` file(s) has been updated:
    - For any service, the `ChangeLog.md` file can be found at `src/{{SERVICE}}/{{SERVICE}}/ChangeLog.md`
    - A snippet outlining the change(s) made in the PR should be written under the `## Upcoming Release` header -- no new version header should be added
- [ ] The PR does not introduce [breaking changes](../blob/master/documentation/breaking-changes/breaking-changes-definition.md)
- [ ] If applicable, the changes made in the PR have proper test coverage
- [x] For public API changes to cmdlets:
    - [x] a cmdlet design review was approved for the changes in [this repository](https://github.com/Azure/azure-powershell-cmdlet-review-pr) (_Microsoft internal only_)
        - {Please put the link here}
    - [x] the markdown help files have been regenerated using the commands listed [here](../blob/master/documentation/development-docs/help-generation.md#updating-all-markdown-files-in-a-module)

### CODE REVIEW

**Review by dingmeng-xue — APPROVED** [review]

**Comment by dingmeng-xue** [comment]
Test case still has problem. Please try to fix it.

If you still hope your feature can go into May 25 release, please change target branch to **release-2021-05-25**

**Comment by hiaga** [comment]
@wyunchi-ms : I've fixed the test cases, Please review this PR.  We hope to catch S187 train

**Comment by hiaga** [comment]
@dingmeng-xue : Could you now review

File: src/RecoveryServices/RecoveryServices/Vault/SetAzureRmRecoveryServicesVaultContext.cs [hunk 1]
```diff
@@ -24,7 +24,6 @@ namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Cmdlets
     /// <summary>
     /// Used to set RecoveryServices Vault Context
     /// </summary>
-    [CmdletDeprecation("This command is getting deprecated. Please use -VaultID parameter in all Recovery Services commands going forward.")]
```
Comment (isra-fel) [inline src/RecoveryServices/RecoveryServices/Vault/SetAzureRmRecoveryServicesVaultContext.cs:27]: Will you confirm you want to withdraw the deprecation of  this cmdlet?

File: src/RecoveryServices/RecoveryServices.Backup.Test/ScenarioTests/AzureFiles/ProtectionCheckTests.cs [hunk 2]
```diff
@@ -21,7 +21,7 @@ namespace Microsoft.Azure.Commands.RecoveryServices.Backup.Test.ScenarioTests
 {
     public partial class ProtectionCheckTests : RMTestBase
     {
-        [Fact]
+        [Fact(Skip = "To un-skip in upcoming release")]
```
Comment (isra-fel) [inline src/RecoveryServices/RecoveryServices.Backup.Test/ScenarioTests/AzureFiles/ProtectionCheckTests.cs:24]: Let me just leave a comment as a reminder to unskip this case

**Review by isra-fel — CHANGES_REQUESTED** [review]
As the merging status suggests: This branch is out-of-date with the base branch
Please sync up with `Azure:release-2021-05-25`. Thanks

File: src/RecoveryServices/RecoveryServices/Vault/SetAzureRmRecoveryServicesVaultContext.cs (hunk 1 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices/Vault/SetAzureRmRecoveryServicesVaultContext.cs:27]: Yes We want to withdraw the deprication, as we are no longer depricating this cmdlet.

File: src/RecoveryServices/RecoveryServices.Backup.Test/ScenarioTests/AzureFiles/ProtectionCheckTests.cs (hunk 2 above)
Comment (hiaga) [reply src/RecoveryServices/RecoveryServices.Backup.Test/ScenarioTests/AzureFiles/ProtectionCheckTests.cs:24]: Ack.

**Review by isra-fel — APPROVED** [review]

**Pull request merged**